packages/core/src/symbols/symbol-flow.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Context,
+  effect,
+  getContext,
+  onCleanup,
+} from "@alloy-js/core/jsx-runtime";
+import { isRef, Ref, shallowReactive } from "@vue/reactivity";
+import { MemberScopeContext, ScopeContext } from "../index.browser.js";
+import { OutputSymbol, OutputSymbolFlags } from "./output-symbol.js";
+
+export interface TakeSymbolCallback {
+  (symbol: OutputSymbol): void;
+}
+
+export interface TakeSymbolsCallback {
+  (symbols: Set<OutputSymbol>): void;
+}
+
+export function takeSymbols() {
+  const context = getContext()!;
+  context.takesSymbols = true;
+  context.takenSymbols = shallowReactive(new Set<OutputSymbol>());
+  return context.takenSymbols;
+}
+
+export function emitSymbol(
+  symbol: OutputSymbol | Ref<OutputSymbol | undefined>,
+) {
+  let ownerContext: Context | undefined;
+  let context = getContext()!.owner;
+  while (context) {
+    if (context.takesSymbols) {
```

Comment:
This is a bit confusing to read. Assigning context to getContext().owner but having `ownerContext` above

Reply:
Renamed to symbolTaker.